src/ensembl/src/content/app/browser/browser-region-editor/BrowserRegionEditor.tsx

```diff
@@ -133,12 +131,15 @@ export const BrowserRegionEditor = (props: BrowserRegionEditorProps) => {
     }
   };
 
-  const changeLocation = (newChrLocation: ChrLocation) =>
-    props.changeBrowserLocation({
-      genomeId: props.activeGenomeId as string,
-      ensObjectId: null,
-      chrLocation: newChrLocation
-    });
+  const changeLocation = (newChrLocation: ChrLocation) => {
+    props.replace(
+      urlFor.browser({
+        genomeId: props.activeGenomeId as string,
+        focus: null,
+        location: newChrLocation ? getChrLocationStr(newChrLocation) : null
+      })
+    );
+  };
```

Comment:
What is wrong with the previous implementation? The previous one should have sent a message to genome browser telling it to go to a different location (see the implementation in browserActions), and when the browser sends a message back, the url should have been updated.

Reply:
Okay, I wasn't aware that the browser was going to send a message to update the URL. With the previous implementation, the RegionEditor kept getting reset which I need to find out why.

Reply:
The loop goes like this:

- we send a bpane event to genome browser: https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/browser/browserActions.ts#L342-L347
- browser changes the content of the canvas and reports back a new location
- we listen to the message from the browser and change the location here: https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/browser/browserActions.ts#L342-L347
- and change the url here: https://github.com/Ensembl/ensembl-client/blob/500e9a4f1969020d9e8290617850202421747c05/src/ensembl/src/content/app/browser/browserActions.ts#L310